src/Seabug/Metadata/Types.purs

```diff
@@ -0,0 +1,299 @@
+            , unsafeMkKey "ownerPkh" /\ toData meta.ownerPkh
+            , unsafeMkKey "ownerPrice" /\ toData meta.ownerPrice
+            ]
+        ]
+    ]
+
+instance FromData SeabugMetadata where
+  fromData sm = unsafePartial do
+    policyId /\ contents <- lookupKey "727" sm >>= case _ of
+      Map [ policyId /\ contents ] ->
+        Tuple <$> fromData policyId <*> fromData contents
+      _ -> Nothing
+    mintPolicy <- lookupKey "mintPolicy" contents >>= fromData
+    collectionNftCS <- lookupKey "collectionNftCS" contents >>= fromData
+    collectionNftTN <- lookupKey "collectionNftTN" contents >>= fromData
+    lockingScript <- lookupKey "lockingScript" contents >>= fromData
+    authorPkh <- lookupKey "authorPkh" contents >>= fromData
+    authorShare <- lookupKey "authorShare" contents >>= fromData
+    marketplaceScript <- lookupKey "marketplaceScript" contents >>= fromData
+    marketplaceShare <- lookupKey "marketplaceShare" contents >>= fromData
+    ownerPkh <- lookupKey "ownerPkh" contents >>= fromData
+    ownerPrice <- lookupKey "ownerPrice" contents >>= fromData
+    pure $ SeabugMetadata
+      { policyId
+      , mintPolicy
+      , collectionNftCS
+      , collectionNftTN
+      , lockingScript
+      , authorPkh
+      , authorShare
+      , marketplaceScript
+      , marketplaceShare
+      , ownerPkh
+      , ownerPrice
+      }
+
+instance DecodeAeson SeabugMetadata where
+  decodeAeson =
+    caseAesonObject
+      (Left (TypeMismatch "Expected object"))
+      $ \o -> do
+          collectionNftCS <-
+            note (TypeMismatch "Invalid ByteArray")
+              <<<
+                ( mkCurrencySymbol <=< hexToByteArray <=<
+                    removeMetadataBytesPrefix
+                )
+              =<< getField o "collectionNftCS"
+          collectionNftTN <-
+            note (TypeMismatch "expected ASCII-encoded `TokenName`")
+              <<< (mkTokenName <=< hexToByteArray <=< removeMetadataBytesPrefix)
+              =<< getField o "collectionNftTN"
+          lockingScript <-
+            map wrap
+              <<< decodeScriptHash =<< getField o "lockingScript"
+          authorPkh <- decodePkh =<< getField o "authorPkh"
+          authorShare <- decodeShare =<< getField o "authorShare"
+          marketplaceScript <- map wrap <<< decodeScriptHash
+            =<< getField o "marketplaceScript"
+          marketplaceShare <- decodeShare =<< getField o "marketplaceShare"
+          ownerPkh <- decodePkh =<< getField o "ownerPkh"
+          ownerPrice <- getField o "ownerPrice"
+          pure $ SeabugMetadata
+            { -- Not used in the endpoints where we parse the metadata, so we
+              -- can set a dummy value
+              policyId: wrap
+                $ unsafePartial
+                $ fromJust
+                $ scriptHashFromBytes
+                $ hexToRawBytesUnsafe
+                    "00000000000000000000000000000000000000000000000000000000"
```

Comment:
Lets make sure we fix this before merging, odd behaviour

Reply:
The issue for this is that it doesn't really make sense to have a `DecodeAeson` instance for `SeabugMetadata`, since the policyId is a key. For an array of `SeabugMetadata` it would make sense, but for `decodeAeson` to work it would need an extra argument specifying the policyId.

I'll update this to remove the `DecodeAeson` instance for now and just use a dedicated `decodeSeabugMetadata` function

Reply:
I also updated the `SeabugMetadata.policyId` field to have type `CurrencySymbol` instead of `MintingPolicyHash` because `CurrencySymbol` is the type we're using for `policyId` everywhere else. I double checked that these get represented in the metadata in the same way with a purchase transaction: the metadata for an NFT before this change:

https://testnet.cardanoscan.io/transaction/dace69f58dc611579bd5c7a7cf0de1d4b307f11f9fce01663bc7799cc6fd799c?tab=metadata

and after this change (after a purchase transaction):

https://testnet.cardanoscan.io/transaction/60c73846b91d7a9ea954211e48b99e8f23ac7c5d989a1d54a31b6e8fdf9b3827?tab=metadata

cardanoscan shows the policyId strangely, but I checked in nami, and you could check in iohk's explorer if it was working.